src/a11y-dialog.ts

```diff
@@ -3,22 +3,27 @@ import focusableSelectors from 'focusable-selectors'
 const TAB_KEY = 'Tab'
 const ESCAPE_KEY = 'Escape'
 
-/**
- * @typedef { import ('./a11y-dialog').A11yDialog } A11yDialogType
- * @typedef { import('./a11y-dialog').EventType } EventType
- * @typedef { import('./a11y-dialog').EventHandler } EventHandler
- * @typedef { import('./a11y-dialog').ListenersRecord } ListenersRecord
- */
+export type A11yDialogEvent = 'show' | 'hide' | 'destroy' | 'create'
+export type A11yDialogInstance = InstanceType<typeof A11yDialog>
+export type A11yDialogEventHandler = (node: Element, event?: Event) => void
+
+type ListenersRecord = Partial<
+  Record<A11yDialogEvent, A11yDialogEventHandler[]>
+>
 
-/** @type A11yDialogType */
 export default class A11yDialog {
-  /** @param {element} HTMLElement */
-  constructor(element) {
+  private $el: HTMLElement
+  private id: string
+  private openers: HTMLElement[]
+  private closers: HTMLElement[]
+  private listeners: ListenersRecord = {}
+  private previouslyFocused: HTMLElement | null = null
+
+  public shown = false
```

Comment:
I’m for the safety of a getter, but to do a getter we’d need some other state from which to derive `shown`, right? unfortunately, I think this property is too simple.

We could mark this as private just to discourage its use, which is the same level of safety as exists now.

Reply:
> This takes the UMD module size up to 1.6kb
My understanding is that it adds ~100 bytes? I believe the UMD to be around 1.5Kb. I would say it’s good this way? Just wondering why you prefixed the property with an underscore. I was under the impression you didn’t like that?

Reply:
> My understanding is that it adds ~100 bytes?

Without using a getter, this code comes in at 1.4kb. We're adding 200 to what we *could* ship and 100 to what we ship now. I defer to you as to whether that's worth it :)

> I was under the impression you didn’t like that

I'm not a fan when we have other ways to tell developers what's private, but I had to make another reference to `shown` so that the getter could return something. This wouldn't work because the property names collide:

``` js
{
  shown: true,
  get shown() { return shown }
}
```

Reply:
Ah yes, I see. Then maybe I‘m overthinking it and keeping it as a simple prop is fine with no getter?

Reply:
I’m fine with that. Since `shown` is public in the currently shipped implementation, we wouldn’t be introducing any new risks. If anything, a getter might deter an author somewhat, but it wouldn’t stop them from setting the secret `_shown` prop if they really wanted to. 

We can mitigate the risk by documenting the importance of `shown` and telling authors not to write to it, maybe?

Reply:
Yes, that sounds like a good plan. Let’s solve that with documentation instead of code. :)